nni/algorithms/compression/pytorch/quantization/quantizers.py

```diff
@@ -545,4 +545,81 @@ def export_model(self, model_path, calibration_path=None, onnx_path=None, input_
 
         self.export_model_save(self.bound_model, model_path, calibration_config, calibration_path, onnx_path, input_shape, device)
 
-        return calibration_config
\ No newline at end of file
+        return calibration_config
+
+
+class LsqQuantizer(Quantizer):
+    """Quantizer defined in:
+       Learned Step Size Quantization (ICLR 2020)
+       https://arxiv.org/pdf/1902.08153.pdf
+       """
+
+    def __init__(self, model, config_list, optimizer=None):
+        super().__init__(model, config_list, optimizer)
+        self.quant_grad = QuantForward()
+        modules_to_compress = self.get_modules_to_compress()
+        for layer, config in modules_to_compress:
+            layer.module.register_parameter("zero_point", torch.nn.Parameter(torch.Tensor([0.0])))
+            layer.module.register_parameter("scale", torch.nn.Parameter(torch.Tensor([1.0])))
+            if "weight" in config.get("quant_types", []):
+                # todo: support per-channel quantization for weight since TensorRT it for conv weight
+                q_bit = get_bits_length(config, "weight")
```

Comment:
In current implementation, we only support single bit quantization in LsqQuantizer? Can we support mixed precision right now?

Reply:
It seems that mixed precision of quantization is supported in this implementation since each layer has its own `q_bit`. We can achieve mixed quantization through some specific settings in `config_list` like：
```
configure_list = [{
        'quant_types': ['weight'],
        'quant_bits': 8,
        'op_types': ['Conv2d'],
        'op_names': ['features.3']
    }, {
        'quant_types': ['weight'],
        'quant_bits': 7,
        'op_types': ['Conv2d'],
        'op_names': ['features.6']
    }]
```